pennylane/tape/qnode.py

```diff
@@ -140,20 +140,28 @@ def __init__(self, func, device, interface="autograd", diff_method="best", **dif
             )
 
         self.func = func
-        self.device = device
+        self._original_device = device
         self.qtape = None
         self.qfunc_output = None
+        # store the user-specified differentiation method
+        self.diff_method = diff_method
 
-        self._tape, self.interface, self.diff_method = self.get_tape(device, interface, diff_method)
+        self._tape, self.interface, diff_method, self.device = self.get_tape(
+            device, interface, diff_method
+        )
+        # The arguments to be passed to JacobianTape.jacobian
         self.diff_options = diff_options or {}
-        self.diff_options["method"] = self.diff_method
+        # Store the differentiation method to be passed to JacobianTape.jacobian().
+        # Note that the tape accepts a different set of allowed methods than the QNode:
+        #     best, analytic, numeric, device

```

Comment:
No problem for this PR, but what is the reason for the difference in method names between the QNode and the Jacobian tape?

Reply:
It's purely historical, they've always deviated 🙁 I wish it were not the case, and wouldn't mind changing it.